packages/fileeditor-extension/src/index.ts

```diff
@@ -9,9 +9,11 @@ import {
 
 import { ICommandPalette, InstanceTracker } from '@jupyterlab/apputils';
 
-import { CodeEditor, IEditorServices } from '@jupyterlab/codeeditor';
+import { CodeEditor } from '@jupyterlab/codeeditor/lib/editor';
```

Comment:
Why are we importing this from `/lib/editor` instead of just the root package?

Reply:
Checking into it - nothing in `/lib/codeeditor` depends on other things in the code editor package, so it's technically valid to import from `/lib/editor`. However, is it necessary and do we want to encourage it? I think our convention is that it's okay to import from `/lib/tokens`. If we start importing from other files like this, it opens up users doing it, which then makes an implicit backwards compatibility issue if we want to refactor things.

Reply:
This might have slipped in as I was testing something. The use of `defaultConfig` does introduce some cross-extension dependency, but I don't really see a clean solution for it (and this doesn't really do anything to help it).